src/Control/Distributed/Spark/SQL/Row.hs

```diff
@@ -1,14 +1,68 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
 
 module Control.Distributed.Spark.SQL.Row where
 
+import Control.Distributed.Closure.TH
 import Control.Distributed.Spark.PairRDD
 import Control.Distributed.Spark.RDD
+import Control.Distributed.Spark.SQL.StructType
+import Data.Int
+import Data.Text
 import Language.Java
 
 newtype Row = Row (J ('Class "org.apache.spark.sql.Row"))
+type instance Interp Row = 'Class "org.apache.spark.sql.Row"
 
 toRows :: PairRDD a b -> IO (RDD Row)
 toRows prdd = callStatic (sing :: Sing "Helper") "toRows" [coerce prdd]
+
+schema :: Row -> IO StructType
+schema (Row r) = call r "schema" []
+
+rowGet :: Int32 -> Row -> IO JObject
```

Comment:
We already have a `Column` module that is intended to be imported qualified. It's only natural that we do the same for `Row` too. This would allow us to stick to the upstream names: `get` and `size` instead of `rowGet` and `rowSize`.